osmnx/core.py

```diff
@@ -787,7 +787,11 @@ def add_edge_lengths(G):
 
     # first load all the edges' origin and destination coordinates as a
     # dataframe indexed by u, v, key
-    coords = np.array([[u, v, k, G.nodes[u]['y'], G.nodes[u]['x'], G.nodes[v]['y'], G.nodes[v]['x']] for u, v, k in G.edges(keys=True)])
+        try:
+        coords = np.array([[u, v, k, G.nodes[u]['y'], G.nodes[u]['x'], G.nodes[v]['y'], G.nodes[v]['x']] for u, v, k in G.edges(keys=True)])
+    except KeyError:
+        missing_nodes = {str(i) for u, v, _ in G.edges(keys=True) if not(G.nodes[u] or G.nodes[u]) for i in (u, v) if not G.nodes[i]}
+        raise TypeError('Edge missing nodes {} likely due to a clipping issue'.format(', '.join(missing_nodes)))
```

Comment:
I would change "likely" to "possibly" as it's difficult to estimate the likelihood representatively across the tool's entire user base.

Reply:
Done plus a reference to "edge(s)" as I found, to my cost, more than one edge may be broken